pkg/kubelet/dockertools/docker.go

```diff
@@ -153,6 +154,14 @@ func filterHTTPError(err error, image string) error {
 
 // Check if the inspected image matches what we are looking for
 func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
+	// NB: the code below this check checks for validity of `image` as a pull spec.
+	// However, it can be useful to also inspect images by ID (config digest, e.g.
```

Comment:
I'm not sure I'm understanding you here, but when you do container-inspect, you get a field called `ImageID`, which we currently use to populate the `ImageID`.  In order to turn that into a manifest digest, you have to do an image-inspect on that config digest to get the information for the image (which includes the `RepoDigests` field).